docs/proxy.md

```diff
@@ -0,0 +1,123 @@
+# Connecting through a proxy
+
+Conneting through a proxy is possible by properly configuring the client constructor and request.
+
+The proxy url should be passed to the client constructor, while the upstream server url
+should be added to every request call in the `path`.
+For instance, if you need to send a request to the `/hello` route of your upstream server,
+the `path` should be `path: 'http://upstream.server:port/hello?foo=bar'`.
+
+If you proxy requires basic authentication, you can send it via the `proxy-authorization` header.
+
+### Connect without authentication
+
+```js
+import { Client } from 'undici'
+import { createServer } from 'http'
+import proxy = from 'proxy'
```

Comment:
this is not valid javascript but typescript.

Reply:
That's actually a typo! :D